### PR TITLE
606 bug fix baseline shodan

### PR DESCRIPTION
**Pull request by jonschlack**
606 bug fix baseline. Thank you.

Some of the changes:

*Removed unused imports
*Added new imports
*Renamed FormValues to FormValuesType
*Added Docstrings to various functions and components that did not have them
*Added code comments to various components
*Removed handleCancel and replaced with [{LoadingOverlayAndCancelButton
*Added Rendercomponent and loadingModal
&Added useEffect function
*Ran Prettier to fix style issues

### CODE REVIEW

**Comment by michaeljpigott** [comment]
Hi @jonschlack 

Can you please use `yarn run style` in your terminal and fix any problems with npx prettier --write <file>? 

Kind regards

Michael

File: src/components/Shodan/Shodan-API-Tool.tsx [hunk 1]
```diff
@@ -1,51 +1,95 @@
...
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [loadingModal, setLoadingModal] = useState(false); // State variable to indicate loading state of the modal.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+
+    // Component Constants.
+    const title = "Shodan API Tool"; // Title of the component.
+    const description =
+        "The Shodan API is a powerful tool that allows external network scans to be performed with use of a valid API key"; // Description of the component.
```
Comment (michaeljpigott) [inline src/components/Shodan/Shodan-API-Tool.tsx:37]: Need a full stop. 
```suggestion
        "The Shodan API is a powerful tool that allows external network scans to be performed with use of a valid API key."; // Description of the component.
```

File: src/components/Shodan/Shodan-API-Tool.tsx [hunk 2]
```diff
@@ -1,51 +1,95 @@
...
+    const steps =
+        "How to use Shodan API:\n" +
+        "Step 1: Install shodkey.py to /usr/share/ddt by running install_exploits.sh script or manually transfer shodkey.py to /usr/share/ddt/\n" +
+        "Step 2: Enter a Valid API Key\n" +
```
Comment (michaeljpigott) [inline src/components/Shodan/Shodan-API-Tool.tsx:41]: Lowercase v. 
```suggestion
        "Step 2: Enter a valid API Key\n" +
```

File: src/components/Shodan/Shodan-API-Tool.tsx [hunk 3]
```diff
@@ -1,51 +1,95 @@
...
+        "Step 3: Enter a Host IP: E.g. 127.0.0.1\n" +
```
Comment (michaeljpigott) [inline src/components/Shodan/Shodan-API-Tool.tsx:42]: Lowercase h. 
```suggestion
        "Step 3: Enter a host IP: E.g. 127.0.0.1\n" +
```

File: src/components/Shodan/Shodan-API-Tool.tsx [hunk 4]
```diff
@@ -1,51 +1,95 @@
...
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
-
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [loadingModal, setLoadingModal] = useState(false); // State variable to indicate loading state of the modal.
```
Comment (michaeljpigott) [inline src/components/Shodan/Shodan-API-Tool.tsx:30]: You may need to change this to true. I am finding that the installation modal briefly pops up each time I go into the tool. Please investigate.

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @jonschlack 

This is looking good. You just need to investigate the InstallationModal because I am finding that it is popping up briefly every time I enter the tool in DDT. 

Kind regards

Michael

File: src/components/Shodan/Shodan-API-Tool.tsx (hunk 4 above)
Comment (jonschlack) [reply src/components/Shodan/Shodan-API-Tool.tsx:30]: > You may need to change this to true. I am finding that the installation modal briefly pops up each time I go into the tool. Please investigate.

Your 100% right. Strange behavior wasn't there last time I ran DDT with updated code. loadingModal needs to be set to true and the popup occurs only first time.. Just compiled and run code with set to true and now it only runs first time. Thank you for your very good help.

**Comment by jonschlack** [comment]
> Hi @jonschlack
> 
> This is looking good. You just need to investigate the InstallationModal because I am finding that it is popping up briefly every time I enter the tool in DDT.
> 
> Kind regards
> 
> Michael

Looked into InstallationModal in Shodan component and changed it to true in component. Appears to have resolved issue of popup every time while entering tool. Please check updated code to see if it's resolved. Thank you!

**Review by michaeljpigott — APPROVED** [review]

**Pull request merged**